### ISSUE TITLE
href.js click handling fires even after e.preventDefault()

### ISSUE THREAD

**Issue by timwis** [yoshuawuyts/sheet-router#50]
`e.preventDefault()` does not seem to stop the click event from bubbling up to `href.js`

```javascript
const choo = require('choo')
const html = require('choo/html')

const app = choo()

const home = (state, prev, send) => html`
	<div>
  	<h1>Home</h1>
  	<a href="/account" onclick=${checkAuth}>Account</a>
  </div>`

const account = (state, prev, send) => html`<div><h1>Account</h1></div>`

app.router((route) => [
  route('/', home),
  route('/account', account)
])

const tree = app.start()
document.body.appendChild(tree)

function checkAuth (e) {
  console.log('attempting to cancel click')
  e.preventDefault()
  // e.stopPropagation() // works when this is also enabled
}
```
Expected behaviour: Clicking on `Account` does nothing (except console log)
Actual behaviour: Clicking on `Account` takes you to `/account`

Currently, you can get around this by using **both** `e.preventDefault()` and `e.stopPropagation()`. I've only tested this in chrome 52.

**Comment by timwis** [comment]
On further reading, I'm not sure this should be unexpected. According to [mdn](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault):

> `preventDefault` doesn't stop further propagation of the event through the DOM. `event.stopPropagation` should be used for that.

Looks like jQuery used [`return false`](https://css-tricks.com/return-false-and-prevent-default/) as a shortcut for calling both.

**Comment by yoshuawuyts** [comment]
What bothers me is that both `preventDefault()` and `stopPropagation()` are needed, I would've expected it to only be the latter.

**Comment by yoshuawuyts** [comment]
Closed as per #247
